spec/feature/authentication_spec.rb

```diff
@@ -0,0 +1,44 @@
+# in spec/features/authentication_spec.rb
```

Comment:
These tests are good. You could have used your web_helpers here to dry out the code.